### PR TITLE
Scope embedded pods to their host targets by their configuration

### PR DESCRIPTION
**Pull request by dnkoutso**
closes https://github.com/CocoaPods/CocoaPods/issues/8011

### CODE REVIEW

File: spec/unit/installer/analyzer_spec.rb [hunk 1]
```diff
@@ -1170,6 +1170,49 @@ module Pod
             JSONKit
             monkey
           ).sort
+          result.targets.flat_map { |at| at.pod_targets_for_build_configuration('Debug').map { |pt| "#{at.name}/Debug/#{pt.name}" } }.sort.should == [
```
Comment (dnkoutso) [inline spec/unit/installer/analyzer_spec.rb]: expanded these tests to be better

File: lib/cocoapods/installer/analyzer.rb [hunk 2]
```diff
@@ -289,7 +289,7 @@ def embedded_target_pod_targets_by_host(aggregate_target, embedded_aggregate_tar
             !aggregate_user_target_uuids.intersection(host_target_uuids).empty?
           end
           embedded_aggregate_target.user_build_configurations.keys.each do |configuration_name|
-            embedded_pod_targets = embedded_aggregate_target.pod_targets.select do |pod_target|
+            embedded_pod_targets = embedded_aggregate_target.pod_targets_for_build_configuration(configuration_name).select do |pod_target|
```
Comment (amorde) [inline lib/cocoapods/installer/analyzer.rb]: hah, nice!

Comment (amorde) [reply lib/cocoapods/installer/analyzer.rb]: I'm assuming this also handles custom build configurations as well?

Comment (dnkoutso) [reply lib/cocoapods/installer/analyzer.rb]: it should! I havent tried it though

**Pull request merged**